spring-beans/src/test/java/org/springframework/beans/factory/support/GenericTypeMatchingTests.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.beans.factory.support;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.core.ResolvableType;
+
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link AbstractBeanFactory#isTypeMatch} with scoped proxy beans that use generic types.
+ */
+class ScopedProxyGenericTypeMatchTests {
+
+	@Test
+	void scopedProxyBeanTypeMatching() {
+		DefaultListableBeanFactory factory = new DefaultListableBeanFactory();
+
+		String targetBeanName = "scopedTarget.wordBean";
+		String proxyBeanName = "wordBean";
```

Comment:
```suggestion
		String proxyBeanName = "wordBean-" + UUID.randomUUID();
		String targetBeanName = "scopedTarget." + proxyBeanName;
```
this should still work, right?